### DISCUSSION TITLE
Help for display module in page product

### DISCUSSION THREAD

**Discussion by Danieleeffe1** [OpenMage/magento-lts#1880] [Questions & Answers]
Goodmorning everyone.

I need help.

I installed a module ([meetanshi wathapp share](https://meetanshi.com/magento-whatsapp-share.html)) that inserts a wathapp button on the product page under the "add to cart" button (reference product.info.addtocart)

I want to move it along with the Twitter and FB "share links".

By modifying the xml file of the module I succeeded in my goal but only in the configurable products.

I have never quite understood the logic behind xml layout files.
I can't find a diagram showing the "references" where to place the modules.

I am forced to go through all the xml files to find the one that refers to the sharing links (which I cannot find) to see how it is written.

How can I do?
Can someone help me?
Thank you.

Below is the layout xml file of the wathapp share module:
```
<?xml version="1.0"?>
<layout version="0.1.0">
    <catalog_product_view>
        <block type="whatsapp/index" name="whatsappimage"  template="whatsapp/catalog/product/view/head.phtml"/>
        <reference name="head" ifconfig="whatsappshare/whatsappconfig/whatsapp_enable">
            <action method="append" ifconfig="whatsappshare/whatsappconfig/whatsapp_enable"><block>whatsappimage</block></action>
            <action method="addCss"><stylesheet>css/whatsapp.css</stylesheet></action>
        </reference>
        <block type="whatsapp/index" name="whatsapp_share" template="whatsapp/catalog/product/view/whatsappshare.phtml" >
            <block type="whatsapp/index" name="display_whatsapp" template="whatsapp/catalog/product/view/displaywhatsapp.phtml"/>
        </block>
        <reference name="product.info.addtocart">
            <action method="append" ifconfig="whatsappshare/whatsappconfig/whatsapp_enable"  ><block>whatsapp_share</block></action>
        </reference>
    </catalog_product_view>
    <catalog_category_default>
        <reference name="head" ifconfig="whatsappshare/whatsappconfig/whatsapp_enable">
            <action method="addCss"><stylesheet>css/whatsapp.css</stylesheet></action>
        </reference>
        <reference name="product_list">
            <action method="setTemplate" ifconfig="whatsappshare/whatsappconfig/whatsapp_enable">
                <template>whatsapp/catalog/product/list.phtml</template>
            </action>
            <block type="whatsapp/index" name="display_list" template="whatsapp/catalog/product/list/displaylist.phtml"/>
            <block type="whatsapp/index" name="display_grid" template="whatsapp/catalog/product/list/displaygrid.phtml"/>
        </reference>
    </catalog_category_default>
</layout>
```

**Comment by luigifab** [comment]
For layout it works like this.
Let's take the example of the contact page with a default installation of OpenMage.

All xml files of enabled modules in `app/design/frontend/rwd/default/layout/*.xml` and all xml files of enabled modules in `app/design/frontend/base/default/layout/*.xml` (only if the file doesn't exist in rwd, for example `page.xml` exists in _rwd_ and _base_, so OpenMage use _page.xml_ of _rwd_ and ignore _page.xml_ of _base_) are **merged** into one single file.

To render the contact page, OpenMage use the `full action name`.
Here it's `contacts_index_index` (read the css classes of _body_ tag).
If you search `<contacts_index_index`  in all xml layout files you will found it in `app/design/frontend/rwd/default/layout/contacts.xml`.
You need also `<default` tags.

As you can see in `<contacts_index_index>`, the layout of the contact page do not `<remove name="root" />` and do not use `output="toHtml"` attribute. So the contact page use `<default>` tag from `app/design/frontend/rwd/default/layout/page.xml`.

The `<default>` define the root block/template: `<block type="page/html" name="root" output="toHtml" template="page/3columns.phtml">`.
This template will be the top of the top of all templates displayed.

But the layout of the contact page says: `<reference name="root"><action method="setTemplate"><template>page/2columns-left.phtml</template></action></reference>`.
So the top of the top of all templates displayed is not `page/3columns.phtml` but `page/2columns-left.phtml`.

If you read `page/2columns-left.phtml`, you will found: `<?php echo $this->getChildHtml('content') ?>`.
This line will output the contact form: `<reference name="content"><block type="core/template" name="contactForm" template="contacts/form.phtml" /></reference>`.

**Reply by luigifab** [reply]
If someone know a good resource that explain that.
